src/coreclr/jit/lower.cpp

```diff
@@ -11145,8 +11145,29 @@ bool Lowering::TryTransformStoreObjAsStoreInd(GenTreeBlk* blkNode)
     GenTree* src = blkNode->Data();
     if (varTypeIsSIMD(regType) && src->IsConstInitVal())
     {
-        // TODO-CQ: support STORE_IND SIMD16(SIMD16, CNT_INT 0).
+#if defined(FEATURE_HW_INTRINSICS) && defined(TARGET_XARCH)
+        assert(!blkNode->ContainsReferences());
+        if (src->OperIsInitVal())
+        {
+            BlockRange().Remove(src);
+            src = src->gtGetOp1();
+        }
+
+        uint8_t initVal = static_cast<uint8_t>(static_cast<size_t>(src->AsIntCon()->IconValue()));
```

Comment:
```suggestion
        uint8_t initVal = static_cast<uint8_t>(src->AsIntCon()->IconValue());
```
Nit, this is unnecessary for a truncating cast